src/parser_urdf.cc

```diff
@@ -3267,44 +3299,41 @@ TiXmlDocument URDF2SDF::InitModelString(const std::string &_urdfStr,
     }
     catch(...)
     {
-      delete sdf;
       throw;
     }
   }
   catch(...)
   {
-    delete robot;
     throw;
```

Comment:
I see the `delete` is removed here and in the previous `catch` block; does this mean there will be memory leaks if either of these `catch` blocks  are executed?

Reply:
It shouldn't be leaked because the corresponding `new` was also removed.  Instead, it's created via the `_sdfXmlOut->NewElement()` call on link 3224.

This is in line with the TinyXML2 documentation:

> An XMLDocument is a C++ object like any other, that can be on the stack, or new'd and deleted on the heap.
> However, any sub-node of the Document, XMLElement, XMLText, etc, can only be created by calling the appropriate XMLDocument::NewElement, NewText, etc. method. Although you have pointers to these objects, they are still owned by the Document. When the Document is deleted, so are all the nodes it contains.

Reply:
👍